src/gridcoinresearchd.cpp

```diff
@@ -85,16 +86,38 @@ bool AppInit(int argc, char* argv[])
             return false;
         }
 
-        LogPrintf("AppInit");
-
         if (!boost::filesystem::is_directory(GetDataDir(false)))
         {
             fprintf(stderr, "Error: Specified directory does not exist\n");
             Shutdown(NULL);
         }
 
+        /** Check here config file incase TestNet is set there and not in mapArgs **/
         ReadConfigFile(mapArgs, mapMultiArgs);
 
+        // Check to see if the user requested a snapshot and we are not running TestNet!
+        if (mapArgs.count("-snapshotdownload") && !mapArgs.count("-testnet"))
+        {
+            Upgrade Snapshot;
+
+            // Let's check make sure gridcoin is not already running in the data directory.
+            // Use new probe feature
+            if (!LockDirectory(GetDataDir(), ".lock", false))
+            {
+                fprintf(stderr, "Cannot obtain a lock on data directory %s.  Gridcoin is probably already running.", GetDataDir().string().c_str());
+
+                exit(1);
+            }
+
+            else
+                Snapshot.SnapshotMain();
+
+            // Delete snapshot file regardless of end result
+            Snapshot.DeleteSnapshot();
```

Comment:
This will not delete the snapshot.zip file if `SnapshotMain()` throws an exception.

Reply:
should catch that.